Extensions/titles.js

```diff
@@ -0,0 +1,104 @@
+//* TITLE Tab titles **//
+//* VERSION 1.0.0 **//
+//* DESCRIPTION Descriptive tab titles, rather than just “Tumblr” **//
+//* DEVELOPER Rebecca Turner **//
+//* FRAME false **//
+//* BETA false **//
+
+XKit.extensions.titles = new Object({
+	running: false,
+
+	/* object mapping the first url folder component to a function which takes
+	 * an array of url components (like ['blog', 'xkit'] for '/blog/xkit') and
+	 * returns the new page title
+	 */
+	titleFuncs: {
+		blog: url => {
```

Comment:
this (and future variables) should probably be named `path` or `pathComponents`, as discussed below.